src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEFieldSymbol.cs

```diff
@@ -581,20 +581,14 @@ public override ImmutableArray<CSharpAttributeData> GetAttributes()
             {
                 var containingPEModuleSymbol = (PEModuleSymbol)this.ContainingModule;
 
-                if (FilterOutDecimalConstantAttribute())
-                {
-                    // filter out DecimalConstantAttribute
-                    var attributes = containingPEModuleSymbol.GetCustomAttributesForToken(
-                        _handle,
-                        out _,
-                        AttributeDescription.DecimalConstantAttribute);
-
-                    ImmutableInterlocked.InterlockedInitialize(ref _lazyCustomAttributes, attributes);
-                }
-                else
-                {
-                    containingPEModuleSymbol.LoadCustomAttributes(_handle, ref _lazyCustomAttributes);
-                }
+                var attributes = containingPEModuleSymbol.GetCustomAttributesForToken(
+                    _handle,
+                    out _,
+                    FilterOutDecimalConstantAttribute() ? AttributeDescription.DecimalConstantAttribute : default,
+                    out _,
+                    IsRequired ? AttributeDescription.RequiredMemberAttribute : default);
```

Comment:
>IsRequired [](http://example.com/codeflow?start=20&length=10)

Given that this is likely to crack attributes, is it worth having it as a condition?

Reply:
Fair question. I was following the existing pattern, but since the presence of the attribute itself is the thing that determines whether or not the member is required, it should be fine to always skip (and even initialize the packaged flag if it wasn't initialized already.